yt_dlp/extractor/twitter.py

```diff
@@ -1640,11 +1644,25 @@ class TwitterBroadcastIE(TwitterBaseIE, PeriscopeBaseIE):
             'upload_date': '20230420',
             'thumbnail': r're:^https?://[^?#]+\.jpg\?token=',
             'view_count': int,
+            'concurrent_view_count': int,
+            'live_status': 'was_live',
         },
+    }, {
+        'url': 'https://x.com/i/events/1910629646300762112',
+        'only_matching': True,
     }]
 
     def _real_extract(self, url):
-        broadcast_id = self._match_id(url)
+        broadcast_type, broadcast_id = self._match_valid_url(url).groups()
+
+        if broadcast_type == 'events':
+            if b_id := traverse_obj(self._call_api(
+                f'live_event/1/{broadcast_id}/timeline.json', broadcast_id,
+            ), ('twitter_objects', 'broadcasts', ..., ('id', 'broadcast_id'), {str_or_none}, any)):
+                return self.url_result(
+                    f'https://x.com/i/broadcasts/{b_id}', TwitterBroadcastIE)
+            raise ExtractorError('No broadcast_id found', expected=True)
```

Comment:
- IMO the assignment expression + API call + traversal done all together is hard to read. I would break this up
- Why return a url result if we can just assign the new/correct value to `broadcast_id`? (rhetorical question; suggestion below)
- In the traversal, we are just branching after `broadcasts` and grabbing the first one? Is this going to be consistently accurate? (actual question; please reply)
- I don't like expecting the error here. We should want a bug report if this breaks

```suggestion
        if broadcast_type == 'events':
            timeline = self._call_api(
                f'live_event/1/{broadcast_id}/timeline.json', broadcast_id)
            broadcast_id = traverse_obj(timeline, (
                'twitter_objects', 'broadcasts', ..., ('id', 'broadcast_id'),
                {str_or_none}, any, {require('broadcast ID')}))
```